Packs/PrismaCloudCompute/Integrations/PaloAltoNetworks_PrismaCloudCompute/PaloAltoNetworks_PrismaCloudCompute.py

```diff
@@ -253,9 +345,529 @@ def fetch_incidents(client):
     return incidents
 
 
+def parse_limit_and_offset_values(limit: str, offset: str) -> Tuple[int, int]:
+    """
+    Parse the offset and limit parameters to integers and verify that the offset/limit are valid.
+
+    Args:
+        limit (str): limit argument.
+        offset (str): offset argument.
+
+    Raises:
+        ValueError: in case the offset/limit values are invalid
+
+    Returns:
+        Tuple[int, int]: parsed offset and parsed limit
+    """
+    limit, offset = arg_to_number(arg=limit, arg_name="limit"), arg_to_number(arg=offset, arg_name="offset")
+
+    try:
+        assert offset is not None and offset >= 0
+        assert limit is not None and 0 < limit <= MAX_API_LIMIT
+    except AssertionError:
+        raise ValueError("limit/offset values are invalid, limit scope = 1-50, offset scope >= 0")
+
+    return limit, offset
+
+
+def parse_date_string_format(date_string: str, new_format: str = "%B %d, %Y %H:%M:%S %p") -> str:
+    """
+    Parses a date string format to a different date string format.
+
+    Args:
+        date_string (str): the date in string representation.
+        new_format (str): the new requested format for the date string.
+
+    Returns:
+        str: date as a new format, in case of a failure returns the original date string.
+    """
+    try:
+        return dateparser.parse(date_string=date_string).strftime(new_format)
+    except AttributeError:
+        return date_string
+
+
+def epochs_to_timestamp(epochs: int, date_format: str = "%B %d, %Y %H:%M:%S %p") -> str:
+    """
+    Converts epochs time representation to a new string date format.
+
+    Args:
+        epochs (int): time in epochs (seconds)
+        date_format (str): the desired format that the timestamp will be.
+
+    Returns:
+        str: timestamp in the new format, empty string in case of a failure
+    """
+    try:
+        return datetime.utcfromtimestamp(epochs).strftime(date_format)
+    except TypeError:
+        return ""
+
+
+def filter_api_response(api_response: list, offset: int, limit: int) -> Optional[list]:
+    """
+    Filter the api response according to the offset/limit, used in case the api doesn't support limit/offset
+
+    Args:
+        api_response (list): api response from an endpoint.
+        offset (int): the offset from which to begin listing the response.
+        limit (int): the maximum limit of records in the response to fetch.
+
+    Returns:
+        list: api filtered response, None in case the api response is empty
+    """
+    if api_response is None:
```

Comment:
is `api_response` can be `None` then its type hint should be `Optional[list]` and better to check `if not api_response`

Reply:
ack